region/tourian/main/Blue Hopper Room.json

```diff
@@ -603,64 +627,89 @@
+      "name": "G-Mode Regain Mobility",
       "requires": [
         {"enemyDamage": {
           "enemy": "Blue Sidehopper",
           "type": "contact",
           "hits": 1
         }}
       ],
-      "exitCondition": {
-        "leaveWithGModeSetup": {}
-      },
-      "unlocksDoors": [{"types": ["ammo"], "requires": []}],
+      "gModeRegainMobility": {},
       "flashSuitChecked": true
     },
     {
-      "link": [2, 2],
-      "name": "G-Mode Setup - Prepared Dodge",
-      "notable": false,
+      "link": [3, 2],
+      "name": "Sidehopper Hit on Entry",
+      "requires": [
+        {"enemyDamage": {
+          "enemy": "Blue Sidehopper",
+          "type": "contact",
+          "hits": 1
+        }}
+      ],
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Sidehopper Prepared Screw Dodge",
       "requires": [
         "canPrepareForNextRoom",
-        {"or": [
-          {"and": [
-            "ScrewAttack",
-            "SpaceJump"
-          ]},
-          {"and": [
-            "Morph",
-            {"or": [
-              "canTrickyJump",
-              {"enemyKill": {
-                "enemies": [["Blue Sidehopper"]]
-              }}
-            ]}
-          ]}
-        ]}
+        "ScrewAttack",
+        "SpaceJump",
+        "canCarefulJump"
+      ],
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Sidehopper Prepared Morph Dodge",
+      "requires": [
+        "canPrepareForNextRoom",
+        "Morph",
+        "canTrickyJump"
       ],
-      "exitCondition": {
-        "leaveWithGModeSetup": {}
-      },
       "unlocksDoors": [
         {"types": ["super"], "requires": []},
         {"types": ["missiles", "powerbomb"], "requires": ["never"]}
       ],
-      "flashSuitChecked": true
+      "note": "Enter the room morphed to avoid a hopper hit on entry."
     },
     {
-      "link": [2, 2],
-      "name": "G-Mode Regain Mobility",
+      "link": [3, 2],
+      "name": "Sidehopper Prepared Morph Dodge Hopper Kill",
       "requires": [
-        {"enemyDamage": {
-          "enemy": "Blue Sidehopper",
-          "type": "contact",
-          "hits": 1
+        "canPrepareForNextRoom",
+        "Morph",
+        "canDodgeWhileShooting",
+        {"enemyKill": {
+          "enemies": [["Blue Sidehopper"]]
         }}
       ],
-      "gModeRegainMobility": {},
-      "flashSuitChecked": true
+      "unlocksDoors": [
+        {"types": ["ammo"], "requires": []}
+      ],
+      "note": "Enter the room morphed to avoid a hopper hit on entry."
     }
-  ]
+  ],
+  "devNote": "FIXME: This room could use obstacles indicating if each door is free of hoppers, allowing opening the door with missiles or a power bomb."
```

Comment:
And so it doesn't use a kill strat into a gmode setup strat.  Usually you can still work around it if that's what logic calls for.